router/router.go

```diff
@@ -260,6 +269,8 @@ func loadConfig() {
 	config.RegisterDurationConfigVariable(48, &failedKeysExpire, true, time.Hour, "Router.failedKeysExpire")
 	config.RegisterDurationConfigVariable(24, &failedKeysCleanUpSleep, true, time.Hour, "Router.failedKeysCleanUpSleep")
 	failedKeysEnabled = config.GetBool("Router.failedKeysEnabled", false)
+
+	config.RegisterBoolConfigVariable(true, &readPerDestination, false, "Router.readPerDestination")
```

Comment:
I will be frank, I don't like that we need to set one extra configuration option for the whole feature to work properly. This is fragile, time passes by and one might forget to turn this on, but enable isolation for some destination type nevertheless.

Reply:
I don't like this either.. I put it as a hack for now, to just test the functionality. I intend to remove this at the end